eo/test/t-eoRankingCached.cpp

```diff
@@ -0,0 +1,156 @@
+#include <eo>
+#include <es/eoReal.h>
+#include <utils/eoRNG.h>
+#include <eoRanking.h>
+#include <apply.h>
+#include "real_value.h"
+
+class RankingTest
+{
+public:
+    RankingTest(eoParser &parser, eoEvalFuncCounter<eoReal<double>> &_eval, unsigned size = 100) : rng(0),
+                                                                                                   popSize(size),
+                                                                                                   seedParam(parser.createParam(uint32_t(time(0)), "seed", "Random seed", 'S')),
+                                                                                                   pressureParam(parser.createParam(1.5, "pressure", "Selective pressure", 'p')),
+                                                                                                   exponentParam(parser.createParam(1.0, "exponent", "Ranking exponent", 'e')),
+                                                                                                   eval(_eval)
+    {
+        rng.reseed(seedParam.value());
+        initPopulation();
+    }
+
+    void initPopulation()
+    {
+        pop.clear();
+        for (unsigned i = 0; i < popSize; ++i)
+        {
+            eoReal<double> ind;
+            ind.resize(1);
+            ind[0] = rng.uniform();
+            pop.push_back(ind);
+        }
+        apply<eoReal<double>>(eval, pop);
+    }
+
+    const unsigned popSize;
+    eoPop<eoReal<double>> pop;
+    eoRng rng;
+    double pressure() const { return pressureParam.value(); }
+    double exponent() const { return exponentParam.value(); }
+
+private:
+    eoValueParam<uint32_t> &seedParam;
+    eoValueParam<double> &pressureParam;
+    eoValueParam<double> &exponentParam;
+    eoEvalFuncCounter<eoReal<double>> eval;
+};
+
+// Test case 1: Verify both implementations produce identical results
+void test_Consistency(eoParser &parser)
+{
+    eoEvalFuncPtr<eoReal<double>, double, const std::vector<double> &> mainEval(real_value);
+    eoEvalFuncCounter<eoReal<double>> eval(mainEval);
+    RankingTest fixture(parser, eval);
+
+    eoRanking<eoReal<double>> ranking(fixture.pressure(), fixture.exponent());
+    eoRankingCached<eoReal<double>> rankingCached(fixture.pressure(), fixture.exponent());
+
+    ranking(fixture.pop);
+    rankingCached(fixture.pop);
+
+    const std::vector<double> &values = ranking.value();
+    const std::vector<double> &cachedValues = rankingCached.value();
+
+    for (unsigned i = 0; i < fixture.pop.size(); ++i)
+    {
+        if (abs(values[i] - cachedValues[i]) > 1e-9)
+        {
+            throw std::runtime_error("Inconsistent ranking values between implementations");
+        }
+    }
+    std::cout << "Test 1 passed: Both implementations produce identical results\n";
```

Comment:
`clog` is more consistent with logging, and `endl` allows for flushing.
```suggestion
    std::clog << "Test 1 passed: Both implementations produce identical results" << std::endl;
```